proto/libp2p.proto

```diff
@@ -123,5 +164,16 @@ message Notification {
         Request request = 2;
         NewPeer new_peer = 3;
         Result result = 4;
+        Join joined = 5;
+        Leave left = 6;
+        Graft grafted = 7;
+        Prune pruned = 8;
+        ValidateMessageGossip validate_message = 9;
+        DeliverMessage deliver_message = 10;
+        UnDeliverableMessage un_deliverable_message = 11;
+        RejectMessage reject_message = 12;
+        DuplicateMessage duplicate_message = 13;
+        AddPeerGossip add_peer = 14;
+        RemovePeerGossip remove_peer = 15;
```

Comment:
`Notification` is getting kind of cramped. We should wrap the tracer messages in a more general one:

```proto
message Notification {
    oneof n {
        ...;
        Tracer tracer = X;
    }
}

message Tracer {
    oneof t {
        Join joined = 1;
        ...;
    }
}
```